src/Compilers/VisualBasic/Portable/Symbols/Source/SourceClonedParameterSymbol.vb

```diff
@@ -225,6 +203,30 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                 End Get
             End Property
 
+            Friend Overrides ReadOnly Property IsCallerLineNumber As Boolean
+                Get
+                    Return _originalParam.IsCallerLineNumber
+                End Get
+            End Property
+
+            Friend Overrides ReadOnly Property IsCallerMemberName As Boolean
+                Get
+                    Return _originalParam.IsCallerMemberName
+                End Get
+            End Property
+
+            Friend Overrides ReadOnly Property IsCallerFilePath As Boolean
+                Get
+                    Return _originalParam.IsCallerFilePath
+                End Get
+            End Property
+
+            Friend Overrides ReadOnly Property CallerArgumentExpressionParameterIndex As Integer
+                Get
+                    Return _originalParam.CallerArgumentExpressionParameterIndex
```

Comment:
>Return _originalParam.CallerArgumentExpressionParameterIndex [](http://example.com/codeflow?start=20&length=60)

It looks like we are effectively clonning another instance of ````SourceClonedParameterSymbol```, whic might have adjusted behavior of CallerArgumentExpressionParameterIndex property, but here we pulling value from the _originalParam of that instance. Is there a guarantee the values are always going to be the same? Including any future implemenations of ```SourceClonedParameterSymbol``` we might add.

Reply:
@AlekseyTs I can think of two approaches here:

1. Creating specialized `*WithCustomModifiers` types for each type that subclasses `SourceClonedParameterSymbol`.
2. Changing the ctor of `SourceClonedParameterSymbolWithCustomModifiers` to accept `SourceClonedParameterSymbol` as `originalParam` (and rename the parameter), then pass it `SourceClonedParameterSymbol` instead of `SourceClonedParameterSymbol._originalParam`.

Let me know which approach do you prefer more, or if both are okay to you I'd probably go for the second.

Reply:
I think this class should use an instance of ```SourceClonedParameterSymbol``` as the original. Might need to change the type of _originalParam in the base